test/e2e/taskdeadline_test.go

```diff
@@ -0,0 +1,182 @@
+		name               string
+		taskDeadline       string
+		expectedCloudEvent eventData
+	}{
+		{
+			name:         "No deadline set - job completes successfully",
+			taskDeadline: "0",
+			expectedCloudEvent: eventData{
+				Project: testEnv.EventData.Project,
+				Result:  "pass",
+				Service: testEnv.EventData.Service,
+				Stage:   testEnv.EventData.Stage,
+				Status:  "succeeded",
+			},
+		},
+		{
+			name:         "Deadline too short - job fails",
+			taskDeadline: "10",
+			expectedCloudEvent: eventData{
+				Project: testEnv.EventData.Project,
+				Result:  "fail",
+				Service: testEnv.EventData.Service,
+				Stage:   testEnv.EventData.Stage,
+				Status:  "errored",
+			},
+		},
+		{
+			name:         "Deadline long enough - job succeeds",
+			taskDeadline: "60",
+			expectedCloudEvent: eventData{
+				Project: testEnv.EventData.Project,
+				Result:  "pass",
+				Service: testEnv.EventData.Service,
+				Stage:   testEnv.EventData.Stage,
+				Status:  "succeeded",
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(
+			test.name, func(t *testing.T) {
+				initialTaskDeadline, err := setConfigMap(
+					testEnv.K8s, testEnv.Namespace, "job-service-config", "task_deadline_seconds",
+					test.taskDeadline,
+				)
+
+				require.NoError(t, err)
+
+				defer func() {
+					setConfigMap(
+						testEnv.K8s, testEnv.Namespace, "job-service-config", "task_deadline_seconds",
+						initialTaskDeadline,
+					)
+					restartJESPod(testEnv.K8s, testEnv.Namespace)
+				}()
+
+				err = restartJESPod(testEnv.K8s, testEnv.Namespace)
+
+				require.NoError(t, err)
+
+				// Send the event to keptn
+				keptnContext, err := testEnv.API.SendEvent(testEnv.Event)
+				require.NoError(t, err)
+
+				// Checking if the job executor service responded with a .started event
+				requireWaitForEvent(
+					t,
+					testEnv.API,
+					2*time.Minute,
+					1*time.Second,
+					keptnContext,
+					"sh.keptn.event.deployment.started",
+					func(_ *models.KeptnContextExtendedCE) bool {
+						return true
+					},
+				)
+
+				requireWaitForEvent(
+					t,
+					testEnv.API,
+					2*time.Minute,
+					1*time.Second,
+					keptnContext,
+					"sh.keptn.event.deployment.finished",
+					func(event *models.KeptnContextExtendedCE) bool {
+						responseEventData, err := parseKeptnEventData(event)
+						require.NoError(t, err)
+
+						t.Log(responseEventData.Message)
+
+						responseEventData.Message = ""
+
+						assert.Equal(t, test.expectedCloudEvent, *responseEventData)
+						return true
+					},
+				)
+			},
+		)
+	}
+
+}
+
+func restartJESPod(clientset *kubernetes.Clientset, namespace string) error {
+	selector := labels.NewSelector()
+	nameSelector, _ := labels.NewRequirement(
+		"app.kubernetes.io/name", selection.Equals, []string{"job-executor-service"},
+	)
+	selector = selector.Add(*nameSelector)
+	pods, err := clientset.CoreV1().Pods(namespace).List(
+		context.TODO(), metav1.ListOptions{LabelSelector: selector.String()},
```

Comment:
I think it should be fine to use `content.Background()` here as it is already used throughout the tests